File: ALittle/Module/ALittleIDE/Script/UIEdit/IDEEditCore/IDESetting/RichEditS.lua
-- ALittle Generate Lua
module("ALittleIDE", package.seeall)

local ___thispath = select('1', ...):match("(.+[/\\]).+$") or ""
local ___rawset = rawset
local ___pairs = pairs
local ___ipairs = ipairs


assert(ALittleIDE.DisplayLayoutS, " extends class:ALittleIDE.DisplayLayoutS is nil")
RichEditS = Lua.Class(ALittleIDE.DisplayLayoutS, "ALittleIDE.RichEditS")

function RichEditS:Ctor(user_info, tab_child, tree_logic)
	___rawset(self, "_layer_name", "ide_setting_richedit")
end

function RichEditS:LoadNatureBase()
	DisplayLayoutS.LoadNatureBase(self)
	self:LoadDefaultNilString("font_path")
	self:LoadDefaultNilString("font_size")
	self:LoadShowTypeData("display_list")
	self:LoadBoolData("editable", true, g_IDEEnum.yn_type)
	self:LoadBoolData("can_drag_text", true, g_IDEEnum.yn_type)
	self:LoadBoolData("focus_cursor", true, g_IDEEnum.yn_type)
	self:LoadDefaultNilString("default_text")
	self:LoadValueData("default_text_alpha")
	self:LoadColorData("cursor_red")
	self:LoadColorData("cursor_green")
	self:LoadColorData("cursor_blue")
	self:LoadColorData("font_red")
	self:LoadColorData("font_green")
	self:LoadColorData("font_blue")
	self:LoadValueData("margin_halign")
	self:LoadValueData("margin_valign")
	self:LoadValueData("cursor_margin_up")
	self:LoadValueData("cursor_margin_down")
	self:LoadValueData("ims_padding")
	self:LoadShowTypeDataForExtends("start_cursor")
	self:LoadShowTypeDataForExtends("end_cursor")
end

function RichEditS:HandleFontPathFOCUSOUT(event)
	self:SpecialFontPathInputChange("font_path", false)
end

function RichEditS:HandleFontSizeFOCUSOUT(event)
	self:ValueNumZInputChange("font_size", false)
end

function RichEditS:HandleFontRedFOCUSOUT(event)
	self:ColorValueInputChange("font_red", false)
end

function RichEditS:HandleFontGreenFOCUSOUT(event)
	self:ColorValueInputChange("font_green", false)
end

function RichEditS:HandleFontBlueFOCUSOUT(event)
	self:ColorValueInputChange("font_blue", false)
end

function RichEditS:HandleCursorRedFOCUSOUT(event)
	self:ColorValueInputChange("cursor_red", false)
end

function RichEditS:HandleCursorGreenFOCUSOUT(event)
	self:ColorValueInputChange("cursor_green", false)
end

function RichEditS:HandleCursorBlueFOCUSOUT(event)
	self:ColorValueInputChange("cursor_blue", false)
end

function RichEditS:HandleDisplayListFOCUSOUT(event)
	if event.target._user_data ~= nil then
		if event.target._user_data == event.target.text then
			return
		end
		event.target._user_data = event.target.text
	end
	self:TableDataSet("display_list", true)
end

function RichEditS:HandleEditableSELECT_CHANGE(event)
	self:BoolSelectChange("editable", false, g_IDEEnum.yn_type)
end

function RichEditS:HandleFocusCurosrSELECT_CHANGE(event)
	self:BoolSelectChange("focus_cursor", false, g_IDEEnum.yn_type)
end

function RichEditS:HandleCanDragTextSELECT_CHANGE(event)
	self:BoolSelectChange("can_drag_text", false, g_IDEEnum.yn_type)
end

function RichEditS:HandleMarginHalignFOCUSOUT(event)
	self:ValueNumInputChange("margin_halign", false)
end

function RichEditS:HandleMarginValignFOCUSOUT(event)
	self:ValueNumInputChange("margin_valign", false)
end

function RichEditS:HandleCursorMarginUpFOCUSOUT(event)
	self:ValueNumInputChange("cursor_margin_up", false)
end

function RichEditS:HandleCursorMarginDownFOCUSOUT(event)
	self:ValueNumInputChange("cursor_margin_down", false)
end

function RichEditS:HandleStartCursorFOCUSOUT(event)
	if event.target._user_data ~= nil then
		if event.target._user_data == event.target.text then
			return
		end
		event.target._user_data = event.target.text
	end
	self:RemoverToNilShowSetForExtends("start_cursor", self._start_cursor.text, false)
end

function RichEditS:HandleEndCursorFOCUSOUT(event)
	if event.target._user_data ~= nil then
		if event.target._user_data == event.target.text then
			return
		end
		event.target._user_data = event.target.text
	end
	self:RemoverToNilShowSetForExtends("end_cursor", self._end_cursor.text, false)
end

function RichEditS:HandleDefaultTextFOCUSOUT(event)
	self:DefaultNoStringInputChange("default_text", false)
end

function RichEditS:HandleDefaultTextAlphaFOCUSOUT(event)
	self:Z2OValueInputChange("default_text_alpha", false)
end

function RichEditS:HandleImsPaddingFOCUSOUT(event)
	self:ValueNumInputChange("ims_padding", false)
end

function RichEditS:HandleRegexFOCUSOUT(event)
	self:DefaultNoStringInputChange("regex", false)
end

function RichEditS:HandleLimitLenFOCUSOUT(event)
	self:ValueNumInputChange("limit_len", false)
end

File: ALittle/Module/ALittleIDE/JSScript/UIEdit/IDEEditCore/IDESetting/RichEditS.js
{
if (typeof ALittleIDE === "undefined") window.ALittleIDE = {};


if (ALittleIDE.DisplayLayoutS === undefined) throw new Error(" extends class:ALittleIDE.DisplayLayoutS is undefined");
ALittleIDE.RichEditS = JavaScript.Class(ALittleIDE.DisplayLayoutS, {
	Ctor : function(user_info, tab_child, tree_logic) {
		this._layer_name = "ide_setting_richedit";
	},
	LoadNatureBase : function() {
		ALittleIDE.DisplayLayoutS.LoadNatureBase.call(this);
		this.LoadDefaultNilString("font_path");
		this.LoadDefaultNilString("font_size");
		this.LoadShowTypeData("display_list");
		this.LoadBoolData("editable", true, ALittleIDE.g_IDEEnum.yn_type);
		this.LoadBoolData("can_drag_text", true, ALittleIDE.g_IDEEnum.yn_type);
		this.LoadBoolData("focus_cursor", true, ALittleIDE.g_IDEEnum.yn_type);
		this.LoadDefaultNilString("default_text");
		this.LoadValueData("default_text_alpha");
		this.LoadColorData("cursor_red");
		this.LoadColorData("cursor_green");
		this.LoadColorData("cursor_blue");
		this.LoadColorData("font_red");
		this.LoadColorData("font_green");
		this.LoadColorData("font_blue");
		this.LoadValueData("margin_halign");
		this.LoadValueData("margin_valign");
		this.LoadValueData("cursor_margin_up");
		this.LoadValueData("cursor_margin_down");
		this.LoadValueData("ims_padding");
		this.LoadShowTypeDataForExtends("start_cursor");
		this.LoadShowTypeDataForExtends("end_cursor");
	},
	HandleFontPathFOCUSOUT : function(event) {
		this.SpecialFontPathInputChange("font_path", false);
	},
	HandleFontSizeFOCUSOUT : function(event) {
		this.ValueNumZInputChange("font_size", false);
	},
	HandleFontRedFOCUSOUT : function(event) {
		this.ColorValueInputChange("font_red", false);
	},
	HandleFontGreenFOCUSOUT : function(event) {
		this.ColorValueInputChange("font_green", false);
	},
	HandleFontBlueFOCUSOUT : function(event) {
		this.ColorValueInputChange("font_blue", false);
	},
	HandleCursorRedFOCUSOUT : function(event) {
		this.ColorValueInputChange("cursor_red", false);
	},
	HandleCursorGreenFOCUSOUT : function(event) {
		this.ColorValueInputChange("cursor_green", false);
	},
	HandleCursorBlueFOCUSOUT : function(event) {
		this.ColorValueInputChange("cursor_blue", false);
	},
	HandleDisplayListFOCUSOUT : function(event) {
		if (event.target._user_data !== undefined) {
			if (event.target._user_data === event.target.text) {
				return;
			}
			event.target._user_data = event.target.text;
		}
		this.TableDataSet("display_list", true);
	},
	HandleEditableSELECT_CHANGE : function(event) {
		this.BoolSelectChange("editable", false, ALittleIDE.g_IDEEnum.yn_type);
	},
	HandleFocusCurosrSELECT_CHANGE : function(event) {
		this.BoolSelectChange("focus_cursor", false, ALittleIDE.g_IDEEnum.yn_type);
	},
	HandleCanDragTextSELECT_CHANGE : function(event) {
		this.BoolSelectChange("can_drag_text", false, ALittleIDE.g_IDEEnum.yn_type);
	},
	HandleMarginHalignFOCUSOUT : function(event) {
		this.ValueNumInputChange("margin_halign", false);
	},
	HandleMarginValignFOCUSOUT : function(event) {
		this.ValueNumInputChange("margin_valign", false);
	},
	HandleCursorMarginUpFOCUSOUT : function(event) {
		this.ValueNumInputChange("cursor_margin_up", false);
	},
	HandleCursorMarginDownFOCUSOUT : function(event) {
		this.ValueNumInputChange("cursor_margin_down", false);
	},
	HandleStartCursorFOCUSOUT : function(event) {
		if (event.target._user_data !== undefined) {
			if (event.target._user_data === event.target.text) {
				return;
			}
			event.target._user_data = event.target.text;
		}
		this.RemoverToNilShowSetForExtends("start_cursor", this._start_cursor.text, false);
	},
	HandleEndCursorFOCUSOUT : function(event) {
		if (event.target._user_data !== undefined) {
			if (event.target._user_data === event.target.text) {
				return;
			}
			event.target._user_data = event.target.text;
		}
		this.RemoverToNilShowSetForExtends("end_cursor", this._end_cursor.text, false);
	},
	HandleDefaultTextFOCUSOUT : function(event) {
		this.DefaultNoStringInputChange("default_text", false);
	},
	HandleDefaultTextAlphaFOCUSOUT : function(event) {
		this.Z2OValueInputChange("default_text_alpha", false);
	},
	HandleImsPaddingFOCUSOUT : function(event) {
		this.ValueNumInputChange("ims_padding", false);
	},
	HandleRegexFOCUSOUT : function(event) {
		this.DefaultNoStringInputChange("regex", false);
	},
	HandleLimitLenFOCUSOUT : function(event) {
		this.ValueNumInputChange("limit_len", false);
	},
}, "ALittleIDE.RichEditS");

}